src/qibo/transpiler/unroller.py

```diff
@@ -1,120 +1,159 @@
-        two_qubit_natives (:class:`qibo.transpiler.abstract.NativeType`): two-qubit native gates
-            supported by the quantum hardware.
+        native_gates (:class:`qibo.transpiler.abstract.NativeGates`): native gates supported by the quantum hardware.
 
     Returns:
         (:class:`qibo.models.circuit.Circuit`): equivalent circuit with native gates.
     """
 
     def __init__(
         self,
-        two_qubit_natives: NativeType,
-        single_qubit_natives=(gates.I, gates.Z, gates.RZ, gates.U3),
-        translate_single_qubit: bool = True,
+        native_gates: NativeGates,
     ):
-        self.two_qubit_natives = two_qubit_natives
-        self.single_qubit_natives = single_qubit_natives
-        self.translate_single_qubit = translate_single_qubit
+        self.native_gates = native_gates
 
     def __call__(self, circuit: Circuit):
-        two_qubit_translated_circuit = circuit.__class__(circuit.nqubits)
         translated_circuit = circuit.__class__(circuit.nqubits)
         for gate in circuit.queue:
-            if len(gate.qubits) > 1 or self.translate_single_qubit:
-                two_qubit_translated_circuit.add(
-                    translate_gate(gate, self.two_qubit_natives)
+            translated_circuit.add(
+                translate_gate(
+                    gate,
+                    self.native_gates,
                 )
-            else:
-                two_qubit_translated_circuit.add(gate)
-        if self.translate_single_qubit:
-            for gate in two_qubit_translated_circuit.queue:
-                if len(gate.qubits) == 1:
-                    translated_circuit.add(translate_gate(gate, self.two_qubit_natives))
-                else:
-                    translated_circuit.add(gate)
-        else:
-            translated_circuit = two_qubit_translated_circuit
+            )
         return translated_circuit
 
+    def is_satisfied(self, circuit: Circuit):
+        """Return True if a circuit is correctly decomposed into native gates, otherwise False."""
+        try:
+            assert_decomposition(circuit=circuit, native_gates=self.native_gates)
+        except DecompositionError:
+            return False
+        return True
+
 
 def assert_decomposition(
     circuit: Circuit,
-    two_qubit_natives: NativeType,
-    single_qubit_natives=(gates.I, gates.Z, gates.RZ, gates.U3),
+    native_gates: NativeGates,
 ):
-    """Checks if a circuit has been correctly decmposed into native gates.
+    """Checks if a circuit has been correctly decomposed into native gates.
 
     Args:
         circuit (:class:`qibo.models.circuit.Circuit`): circuit model to check.
+        native_gates (:class:`qibo.transpiler.abstract.NativeGates`):
+            native gates supported by the quantum hardware.
     """
     for gate in circuit.queue:
         if isinstance(gate, gates.M):
             continue
         if len(gate.qubits) == 1:
```

Comment:
I think here you can simplify by handling the one- and two-qubit gates together. In the end you just need to check if 
```py
NativeGates.from_gate(gate) & native_gates
```
is non-empty. This check works in both cases.

It is also fine to raise the same error message. The gate name appears there so the user already knows if it's one or two qubit.